### DISCUSSION TITLE
How do I limit the number of players per party?

### DISCUSSION THREAD

**Discussion by b3aribeiro** [jbakse/p5.party#46] [Q&A] [answered]
I want to develop microgames where I need to limit the number of players per party. 
Is there a way of doing it? 

Then, let's say I reach the max amount in a room... how do I assign extra players an "observer mode"?

**Comment by jbakse** [comment, marked as answer]
Take a look at the pong example, and see if that is helpful.

If not it will probably help to have a more detailed specific example of one thing you might want to do in this area.

**Reply by b3aribeiro** [reply]
Not sure if it's a bug, but when I open a third window for [pong](https://p5party.netlify.app/show_example.html?pong_m) it freezes the screen for the observer. 
I want the observer to watch in real-time what's happening in the game without. Is that possible?

**Reply by jbakse** [reply]
Yeah, that sounds like a bug. That example should pick one client to be "player 1", one to be "player 2", and all others should be "observer". Observer should see the action, but won't control a paddle.

That much is working for me now, but i do see "state transition" errors in the console when a client reloads, I'll look into that.

Please take a look at the example again, and if the observer screen is frozen for you, let me know what browser you are using, or any other hints why it might work differently for you.

**Reply by b3aribeiro** [reply]
I tested again and I notice that this glitch only happens to me if open the tabs in the same window. 

I'm using google chrome and when I tried to open the game in three different google chrome windows, I didn't encounter any problems. 
However, if I open the tabs in the same window, it causes the game to crash - even for the second window

Here's how my console looks like when I try to open two tabs in the same window. 


First I open TAB 1:
> debug connectionStateChanged AWAITING_CONNECTION
> Client.js:27 debug connectionStateChanged CHALLENGING
> Client.js:27 debug connectionStateChanged AWAITING_AUTHENTICATION
> Client.js:27 debug connectionStateChanged AUTHENTICATING
> Client.js:27 debug connectionStateChanged OPEN
> Record.js:97 debug RecordManager: Record ready. drag-main/shared
> index_p5.js:61 party partyLoadShared done! shared
> Record.js:97 debug RecordManager: Record ready. drag-main/_kz4z2w0u-1d0tlw6dv4c
> index_p5.js:82 party partyLoadMyShared done!
> index_p5.js:42 party partyConnect done!
> Record.js:97 debug RecordManager: Record ready. drag-main/_kz4z2nd3-1d6b11fesz5
> index_p5.js:105 party partyLoadParticipantShareds done!

Then I open TAB 2.
> debug connectionStateChanged AWAITING_CONNECTION
> Client.js:27 debug connectionStateChanged CHALLENGING
> Client.js:27 debug connectionStateChanged AWAITING_AUTHENTICATION
> Client.js:27 debug connectionStateChanged AUTHENTICATING
> Client.js:27 debug connectionStateChanged OPEN
> Record.js:97 debug RecordManager: Record ready. drag-main/shared
> index_p5.js:61 party partyLoadShared done! shared
> Record.js:97 debug RecordManager: Record ready. drag-main/_kz4z2w0u-1d0tlw6dv4c
> index_p5.js:82 party partyLoadMyShared done!
> index_p5.js:42 party partyConnect done!
> Record.js:97 debug RecordManager: Record ready. drag-main/_kz4z2nd3-1d6b11fesz5
> index_p5.js:105 party partyLoadParticipantShareds done!


Then I went back to TAB 1 to check the console.
> Record.js:97 debug RecordManager: Record ready. drag-main/_kz4z2w0u-1d0tlw6dv4c


Nothing else happened for tab 2's console.
Also, I could move my paddle for tab 2, but the ball and enemy's paddle was frozen.
For tab1, the game seemed normal.

I hope that helps.

**Reply by jbakse** [reply]
Ah, could it be this:
If a p5 sketch is in a tab that isn't showing it is "paused". If the host isn't in a visible window, its going to be paused. So your observer might not be frozen, its seeing that host is paused? 
If that is what is happening opening all the tabs in their own windows so they can all run should resolve it.

**Reply by b3aribeiro** [reply]
Exactly!  "Open this example in two separate browser **windows** (not tabs) at once!"
